### PR TITLE
feature/issue 165/internal group detail view

### LINKED ISSUE #64: InternalGroup and InternalGroupPosition
[Looking at the InternalGroup and InternalGroupPosition models](https://github.com/KSG-IT/ksg-nett/blob/develop/organization/models.py), I think perhaps they should be rewritten slightly. I am not actually sure that it makes sense to limit the names by choices. Or put differently, I am not sure if the possible groups should be limited from the code-side.

Also, `InternalGroupPosition` should potentially be directly related to _one_ `InternalGroup`.

I am assigning this task to myself, as I am due to start implement the frontend pages related to these (and similar models). This issue should be potentially resolved before the frontend is implemented.

### LINKED ISSUE #165: Create internal group detail view
A detail view listing information of an internal group

### PR DESCRIPTION
**Pull request by alexaor**
Closes #165, closes #64 

A first draft at how to handle modelling of internal groups and commissions. This should cover maybe most bases? Open for discussion

Also worth mentioning it alters a very old migration file in organization so in order for this to not break you will have to migrate backwards before pulling, pull, and then migrate. The migration in question is the third one in the organization module.

[Github gist](https://gist.github.com/alexaor/730b7b0184336e575f20d8a4bd8ed722) with data dumps which hopefully can quickly be loaded. User profile images probably has to be rewritten though

Commit summary:
- refactor(organization): migrate from model_utils in models
- feat(organization): move model charfield choices to consts file
- feat(organization): add type and alter name in InternalGroup model
- feat(organization): alter InternalGroupPosition model
- refactor: migrate Comission model to organization app
- feat(org): base organization module model structure
- feat(org): flesh out admin panel to reflect recent model changes
- feat: move organization urls inside internal view
- fix(core): fix header squish on small screen
- feat(org): internal-group detail view bootstrap

### CODE REVIEW

File: organization/consts.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+
```
Comment (alexaor) [inline organization/consts.py]: Should probably just straight up delete the whole file

File: commissions/models.py [hunk 2]
```diff
@@ -1,16 +1,3 @@
 from django.db import models
```
Comment (alexaor) [inline commissions/models.py:1]: Wanted to delete the whole app in the same PR but due to some migration dependency headaches in users it was put on hold

File: organization/admin.py [hunk 3]
```diff
@@ -3,7 +3,73 @@
...
+        "name",
+        "active_members"
+    )
+
+    inlines = (
+        CommissionMembershipInline,
+    )
+
+    def active_members(self, obj: Commission):
+        return obj.memberships.filter(date_ended__isnull=True).count()
```
Comment (alexaor) [inline organization/admin.py]: This should probably come from a property within the model definition itself

File: organization/migrations/0008_auto_20201114_1556.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 3.1.2 on 2020-11-14 15:56
```
Comment (alexaor) [inline organization/migrations/0008_auto_20201114_1556.py:1]: Consider squashing all the new migration files?

File: organization/models.py [hunk 5]
```diff
@@ -2,105 +2,145 @@
...
-    )
+    class Type(models.TextChoices):
+        """
+        Making use of https://docs.djangoproject.com/en/3.1/ref/models/fields/#enumeration-types
+        Denotes the internal group type, either a interest-group, e.g. KSG-iT, Påfyll etc. or an internal
+        grouping, e.g. Lyche bar og Edgar
+        """
+        INTEREST_GROUP = "interest-group"
+        INTERNAL_GROUP = "internal-group"
+
```
Comment (alexaor) [inline organization/models.py:23]: Have an image field which is rendered as a banner image over the description of the internal group if it exists?

File: organization/models.py [hunk 6]
```diff
@@ -2,105 +2,145 @@
...
+    name = models.CharField(unique=True, max_length=32)
+    type = models.CharField(max_length=32, null=False, blank=False, choices=Type.choices)
+    description = models.TextField(max_length=2048, blank=True, null=True)
+
+    @property
+    def slugify(self):  # is this stupid way to do it? Just use ID's instead?
+        return "-".join(self.name.lower().split(" ")).replace("ø", "o").replace("æ", "ae").replace("å", "a")
```
Comment (alexaor) [inline organization/models.py]: Moved away from having the name of the internal group as a slug in the URL and instead just used the ID of the object. Should be deleted.

File: organization/models.py [hunk 7]
```diff
@@ -2,105 +2,145 @@
...
+    date_joined = models.DateField(auto_now_add=True)
+    date_ended = models.DateField(
+        default=None,
+        null=True,
+        blank=True
+    )
+
+    user = models.ForeignKey("users.User", on_delete=models.CASCADE)
+    position = models.ForeignKey("organization.InternalGroupPosition", related_name="memberships",
+                                 on_delete=models.DO_NOTHING)
```
Comment (alexaor) [inline organization/models.py]: Is it more correct with CASCADE here? I reckon that a internal group should never be deleted as long as there are members. Maybe CASCADE here and DO_NOTHING on the position model?

File: organization/models.py [hunk 8]
```diff
@@ -2,105 +2,145 @@
...
+    # Add a flag for `in_charge` or something? Does this problem actually need to be solved programaticall
+
+
+class Committee(models.Model):
+    # not sure if this model makes sense whatsoever. Deal with this later
+
+    members = models.ManyToManyField(
+        Commission,
+        related_name="committees"
+    )
```
Comment (alexaor) [inline organization/models.py:157]: Poor attempt to have a model for extraordinary groupings related to commissions. Things like DM, or other committees which might exist in the future as the orgniazation grows

File: organization/views.py [hunk 9]
```diff
@@ -1,5 +1,32 @@
...
-from __future__ import unicode_literals
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import render, HttpResponse
+from organization.models import InternalGroup
+from rest_framework import status
 
 
-# Create your views here.
+@login_required
+def internal_group(request, internal_group_id):
```
Comment (alexaor) [inline organization/views.py]: should this be renamed `internal_group_detail`? Feels like its a bit too close to the naming when we make a general landing page which would probably be named `internal_groups`.

File: users/models.py [hunk 10]
```diff
@@ -72,14 +71,14 @@ class User(AbstractUser):
...
-        on_delete=models.SET_NULL
-    )
+    # commission = models.ForeignKey(
+    #     "organization.Commission",
+    #     default=None,
+    #     blank=True,
+    #     null=True,
+    #     related_name='holders',
+    #     on_delete=models.SET_NULL
+    # )
```
Comment (alexaor) [inline users/models.py]: Remove

**Comment by alexaor** [comment]
[Github gist](https://gist.github.com/alexaor/730b7b0184336e575f20d8a4bd8ed722) with data dumps which hopefully can quickly be loaded. User profile images probably has to be rewritten though

File: organization/models.py (hunk 5 above)
Comment (alexaor) [reply organization/models.py:23]: will instead have this as some future enhancement feature

File: organization/models.py (hunk 7 above)
Comment (trmdy) [reply organization/models.py]: I feel like `CASCADE` is the way to go here? When a position is deleted, clearly the membership to that position should be deleted, yes?

File: organization/models.py [hunk 11]
```diff
@@ -2,105 +2,153 @@
...
     """
-    STATUS = KSG_POSITIONS
 
-    name = StatusField(unique=True)
+    class Meta:
+        unique_together = ("name", "internal_group",)
 
-    description = models.CharField(max_length=1024, blank=True, null=True)
+    # feels natural to have the type here, like functionary or gang-member, but how do we deal with it as an
+    # interest group? Do we just have null=True and then not set it?
```
Comment (trmdy) [inline organization/models.py]: I guess the enum can just be exhaustive here? I.e. let it be

* Functionary
* Gang-member
* Interest-group-member
* Active functionary pang
* Old functionary pang
* Active gang-member pang
* Old gang-member pang
* Hangaround

File: organization/models.py [hunk 12]
```diff
@@ -2,105 +2,153 @@
...
+
+
+class Commission(models.Model):
+    """
+    A commissions (verv) in KSG. A commissions can be shared by many users (e.g. Personal),
+    or created specifically for this internal group (e.g. Hybelbarsjef).
+    """
+
+    name = models.CharField(max_length=32, unique=True)
+    manager = models.ForeignKey(  # does this thing even make sense to have?
```
Comment (trmdy) [inline organization/models.py]: The idea is that a single person should manage the commission?

File: organization/models.py [hunk 13]
```diff
@@ -2,105 +2,153 @@
...
+        return "Commission(name=%s)" % (self.name,)
+
+
+class CommissionMembership(models.Model):
+    user = models.ForeignKey("users.User", null=False, on_delete=models.CASCADE, related_name="commission_history")
+    commission = models.ForeignKey("organization.Commission", on_delete=models.DO_NOTHING, related_name="memberships")
+    date_started = models.DateField(auto_now_add=True)
+    date_ended = models.DateField(default=None, null=True, blank=True)
+
+    # What happens if a person is personal first semester and is it again their last semester? Just spawn a
```
Comment (trmdy) [inline organization/models.py]: Yeah, a new instance for sure. It makes more sense for history as well.

**Review by trmdy — CHANGES_REQUESTED** [review]
Looks like a good start. Added some comments you can start working on. I'll deal with the frontend after that.

File: organization/models.py (hunk 12 above)
Comment (alexaor) [reply organization/models.py]: I suppose in some cases where it makes sense, but not very happy with this approach. Think I'll scrap this field for the time being.

**Review by trmdy — APPROVED** [review]
LGTM

**Pull request merged**